### PR TITLE
cohttp-eio: generate Date header in Server

### PR DESCRIPTION
**Pull request by bikallem**
Generates Date header in responses from `Cohttp_eio.Server`. 

Fixes https://github.com/mirage/ocaml-cohttp/issues/932

/cc @mseri @talex5

### CODE REVIEW

File: cohttp-eio/src/cohttp_eio.mli [hunk 1]
```diff
@@ -84,15 +92,14 @@ module Server : sig
   (** {1 Run Server} *)
 
   val run :
-    ?socket_backlog:int ->
-    ?domains:int ->
-    port:int ->
-    < domain_mgr : Eio.Domain_manager.t ; net : Eio.Net.t ; .. > ->
-    handler ->
-    'a
+    ?socket_backlog:int -> ?domains:int -> port:int -> 'b env -> handler -> 'a
```
Comment (mseri) [inline cohttp-eio/src/cohttp_eio.mli:95]:
```suggestion
    ?socket_backlog:int -> ?domains:int -> port:int -> 'a env -> handler -> 'b
```
Pointless comment, but why not just using the same as below

File: cohttp-eio/src/server.ml [hunk 2]
```diff
@@ -65,14 +73,48 @@ let internal_server_error_response =
...
+    | 6 -> "Jun"
+    | 7 -> "Jul"
+    | 8 -> "Aug"
+    | 9 -> "Sep"
+    | 10 -> "Oct"
+    | 11 -> "Nov"
+    | 12 -> "Dec"
+    | _ -> failwith "Invalid HTTP datetime value"
+  in
+  Format.sprintf "%s, %02d %s %04d %02d:%02d:%02d GMT" weekday dd month year hh
```
Comment (mseri) [inline cohttp-eio/src/server.ml:106]: The specification is https://www.rfc-editor.org/rfc/rfc5322#section-3.3

If I am not mistaken `to_date_time` will give the time as UTC/GMT, is that what the env#clock generates or we need to be careful with the timezone?

File: cohttp-eio/src/server.ml [hunk 3]
```diff
@@ -65,14 +73,48 @@ let internal_server_error_response =
...
+
+let write_response ?request env writer (response, body) =
   let headers =
     let request_meth = Option.map Http.Request.meth request in
     Body.add_content_length
       (Http.Response.requires_content_length ?request_meth response)
       (Http.Response.headers response)
       body
   in
+  let headers = Http.Header.add headers "Date" (http_date env) in
```
Comment (mseri) [inline cohttp-eio/src/server.ml:117]: Do we want to make it conditional only on the status being `2xx`, `3xx`, `4xx`?
We could save on some unnecessary clock calls

File: cohttp-eio/src/server.ml (hunk 2 above)
Comment (bikallem) [reply cohttp-eio/src/server.ml:106]: Yes. `Eio.Time.now` uses `Unix.gettimeofday()` so the time is in UTC timezone.

File: cohttp-eio/src/cohttp_eio.mli (hunk 1 above)
Comment (bikallem) [reply cohttp-eio/src/cohttp_eio.mli]: Done.

File: cohttp-eio/src/server.ml (hunk 3 above)
Comment (bikallem) [reply cohttp-eio/src/server.ml]: Good suggestion. Date header generation is now excluded on Informational and Server error responses.

**Comment by bikallem** [comment]
@mseri I have addressed all of your feedback.

**Pull request merged**